templates/KnockoutSpa/ClientApp/components/app-root/app-root.ts

```diff
@@ -40,4 +40,4 @@ class AppRootViewModel {
     }
 }
 
-export default { viewModel: AppRootViewModel, template: require('./app-root.html') };
+export default { viewModel: AppRootViewModel, componentName: 'Home', template: require('./app-root.html') };
```

Comment:
Sounds like the URL isn't matching any route, so `route()` is an empty object, so `route().page` is undefined, so you're returning nothing for the `name` of the component.